tests/integration/models/application.spec.js

```diff
@@ -661,6 +661,18 @@ describe('Application Model', function () {
 					}),
 				);
 
+				applicationRetrievalFields.forEach((prop) =>
+					it(`should be able to generate a provisioning key by ${prop} with key name as key_${prop}`, async function () {
+						const key = await balena.models.application.generateProvisioningKey(
+							this.application[prop],
+							`key_${prop}`,
+						);
+
+						expect(key).to.be.a('string');
+						expect(key).to.have.length(32);
```

Comment:
> Can we also get the latest api key that we have for this app and check it's name property?
We can also store the last known api key id and use it in the filter so that we can be more confident that the one that we got back is the one that we just created.